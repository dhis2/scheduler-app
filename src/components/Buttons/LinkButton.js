import React from 'react'
import { func, node, string, oneOfType } from 'prop-types'
import styles from './LinkButton.module.css'

// List of the props that should not be passed on to the link
const filteredPropList = [
    'primary',
    'secondary',
    'destructive',
    'small',
    'large',
]

/**
 * Allows you to render react-router links, or regular links as buttons, so they look similar to
 * our buttons. Props that are used by the button will not be passed on to the link.
 */

const LinkButton = ({ as: Element, ...rest }) => {
    const classNames = [styles.wrapper]

    filteredPropList.forEach(prop => {
        if (rest[prop]) {
            classNames.push(styles[prop])
            delete rest[prop]
        }
    })

    return <Element className={classNames.join(' ')} {...rest} />
}

LinkButton.propTypes = {
    as: oneOfType([string, func]).isRequired,
    children: node.isRequired,
}

export default LinkButton
