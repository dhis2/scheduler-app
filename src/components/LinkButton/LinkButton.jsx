import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import { Link } from 'react-router-dom'
import styles from './LinkButton.module.css'

const LinkButton = ({ className, small, ...rest }) => {
    return (
        <Link
            className={cx(
                styles.linkButton,
                { [styles.small]: small },
                className
            )}
            {...rest}
        />
    )
}

LinkButton.propTypes = {
    className: PropTypes.string,
    small: PropTypes.bool,
}

export default LinkButton
