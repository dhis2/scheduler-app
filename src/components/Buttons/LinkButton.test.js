import React from 'react'
import { shallow } from 'enzyme'
import LinkButton from './LinkButton'

describe('<LinkButton>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<LinkButton as="a">Link text</LinkButton>)

        expect(wrapper).toMatchSnapshot()
    })

    it('does not pass filtered props to the link', () => {
        const wrapper = shallow(
            <LinkButton primary as="a">
                Link text
            </LinkButton>
        )

        expect(wrapper).toMatchSnapshot()
    })
})
