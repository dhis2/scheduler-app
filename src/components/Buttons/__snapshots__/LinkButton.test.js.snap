// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton> does not pass filtered props to the link 1`] = `
<a
  className="wrapper primary"
>
  Link text
</a>
`;

exports[`<LinkButton> renders correctly 1`] = `
<a
  className="wrapper"
>
  Link text
</a>
`;
